pkg/client/unversioned/deployment_test.go

```diff
@@ -124,6 +124,27 @@ func TestDeploymentUpdate(t *testing.T) {
 	c.Validate(t, response, err)
 }
 
+func TestDeploymentUpdateStatus(t *testing.T) {
+	ns := api.NamespaceDefault
+	deployment := &extensions.Deployment{
+		ObjectMeta: api.ObjectMeta{
+			Name:            "abc",
+			Namespace:       ns,
+			ResourceVersion: "1",
+		},
+	}
+	c := &testClient{
+		Request: testRequest{
+			Method: "PUT",
+			Path:   testapi.Extensions.ResourcePath(getDeploymentsResoureName(), ns, "abc") + "/status",
+			Query:  buildQueryValues(nil),
+		},
+		Response: Response{StatusCode: 200, Body: deployment},
+	}
+	response, err := c.Setup(t).Deployments(ns).UpdateStatus(deployment)
+	c.Validate(t, response, err)
```

Comment:
ok. I think those would be tests in registery/deployment/etcd/. These client tests don't actually pass through registry IIUC.